packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/rendering.dart';
+
+@visibleForTesting
+const double kDefaultTableViewCellHeight = 44.0;
+
+@visibleForTesting
+const double kDefaultCellInset = 16.0;
+
+@visibleForTesting
+const double kDefaultDividerHeight = 1.0;
+
+/// WARNING: CupertinoTableView and its associated classes are EXPERIMENTAL. IF
+/// YOU USE THESE APIS, WE WILL BREAK YOU!
+///
+/// An iOS-styled table view.
+///
+/// Table views are used in iOS to render lists of cells. They are the iOS
+/// equivalent of Android's `ListView` and `RecyclerView`. [CupertinoTableView]
+/// is Flutter's iOS-styled alternative to [ListView].
+///
+/// #Cells
+///
+/// To achieve complete iOS style parity, consider filling [CupertinoTableView]s
+/// with [CupertinoTableViewCell]s. If custom cells are desired, any widget
+/// can be added to a [CupertinoTableView].
+///
+/// #Dividers
+///
+/// By default, [CupertinoTableView]s render dividers between each cell. For
+/// custom cells, the divider beneath the cell is rendered with a standard inset
+/// on the starting edge and is then drawn all the way to the ending edge. For
+/// [CupertinoTableViewCell]s, the divider starts where the title text starts
+/// and then renders all the way to the ending edge. This divider behavior is
+/// as per iOS behavior.
+///
+/// If a [CupertinoTableView] does not contain enough cells to fill the viewport
+/// then extra dividers are rendered beyond the last cell to the end of the
+/// viewport. This extra divider rendering extends into overscroll space at the
+/// end, too. The gap between extra dividers is equal to the height of the
+/// last cell in the [CupertinoTableView].
+///
+/// #Styles
+///
+/// A [CupertinoTableView] may be rendered either in a "plain" style or a
+/// "grouped" style.
+///
+/// A plain [CupertinoTableView] renders a single set of cells without any
+/// divisions between cells.
+///
+/// A grouped [CupertinoTableView] subdivides cells into sections. Sections
+/// are rendered with gaps between them and optional headers before each section.
+///
+/// #Section Index
+///
+/// If a [CupertinoTableView] is in a grouped style, it can optionally display
+/// a section "index". This index is a thin strip that appears on the right side
+/// of the [CupertinoTableView], displays titles for each section in the
+/// [CupertinoTableView], and allows users to tap on a given section title to
+/// automatically scroll to that section.
+///
+/// See also:
+///
+///  * [CupertinoTableViewCell], which renders an iOS-style table view cell that
+///    is intended to appear within a [CupertinoTableView].
+///  * [ListView], which renders a style agnostic list.
+class CupertinoTableView extends StatefulWidget {
```

Comment:
I don't think we need the verbosity since almost everything is a 'view'. 

![cncb09ovmaaqvel](https://user-images.githubusercontent.com/156888/48864332-11343b80-ed81-11e8-9afc-b350d67a627f.jpg)

Reply:
I'm following the iOS name with the "UI" replaced by "Cupertino", I think that's a better rule of thumb of all such widgets.

Reply:
IMHO this should be CupertinoTableView (or CupertinoListView) but for entirely different reasons: I think it should inherit from ScrollView (directly or indirectly through BoxScrollView), just like ListView, and all ScrollView subclasses end with "View".

If it was called CupertinoTable, then I would expect it to be a Table subclass.

Reply:
Discussed offline yesterday and SG if we subclass BoxScrollView.

Reply:
(I don't think we need to worry about matching iOS terminology in the class names. We're never going to have perfect API parity, and we shouldn't aim for it, because we want our API to be much easier to use. We should just have documentation that maps iOS names to our names and make sure we're consistent within our own API on the naming. That would, for example, argue for a CupertinoListView rather than a CupertinoTableView. At the end of the day, success for us looks like more of our developers writing iOS apps with our cupertino library without having ever written directly with UiKit than having first used UiKit. In other words, success for us is us growing the pie of iOS app developers, not moving developers from UiKit to Flutter.)